src/models/session.model.ts

```diff
@@ -1,45 +1,45 @@
 import i18n from "i18next";
 
 import { Session as ProtoSession } from "@ak-proto-ts/sessions/v1/session_pb";
-import { SessionEntrypoint, ViewerSession } from "@src/types/models/session.type";
-import { EntrypointTrigger, Session, Value } from "@type/models";
+import { Session, ViewerSession } from "@src/interfaces/models";
+import { Value } from "@src/types/models";
 import { convertTimestampToDate, parseNestedJson } from "@utilities";
-/**
- * Converts a ProtoSession object to a SessionType object.
- * @param protoSession The ProtoSession object to convert.
- * @returns The SessionType object.
- */
-export function convertSessionProtoToModel(protoSession: ProtoSession): Session {
+
+function convertProtoSessionBase(protoSession: ProtoSession) {
 	return {
-		createdAt: convertTimestampToDate(protoSession.createdAt!),
-		deploymentId: protoSession.deploymentId,
-		entrypoint: protoSession.entrypoint as unknown as EntrypointTrigger,
+		createdAt: convertTimestampToDate(protoSession.createdAt),
 		inputs: parseNestedJson(protoSession.inputs as Value),
 		sessionId: protoSession.sessionId,
 		state: protoSession.state,
-		triggerName: protoSession.memo.trigger_name,
-		connectionName: protoSession.memo.connection_name,
+		triggerName: protoSession.memo?.trigger_name,
+		entrypoint: {
+			col: protoSession.entrypoint?.col,
+			name: protoSession.entrypoint?.name,
+			path: protoSession.entrypoint?.path,
+			row: protoSession.entrypoint?.row,
+		},
```

Comment:
why not  `entrypoint: protoSession.entrypoint` ?

Reply:
`protoSession.entrypoint` type is **CodeLocation**
`session.entrypoint` type is **SessionEntrypoint**

I improved the code to this:
```
		entrypoint: { ...protoSession.entrypoint },
```